src/Interpreters/FunctionNameNormalizer.cpp

```diff
@@ -0,0 +1,41 @@
+#include <Interpreters/FunctionNameNormalizer.h>
+
+#include <Parsers/ASTColumnDeclaration.h>
+#include <Parsers/ASTCreateQuery.h>
+
+namespace DB
+{
+
+const String & getFunctionCanonicalNameIfAny(const String & name);
+const String & getAggregateFunctionCanonicalNameIfAny(const String & name);
+
+void FunctionNameNormalizer::visit(IAST * ast)
+{
+    if (!ast)
+        return;
+
+    if (auto * node_storage = ast->as<ASTStorage>())
```

Comment:
Btw, why this code path is needed, and 

```
    for (auto & child : ast->children)
        visit(child.get());
```

is not enough?

Reply:
It normalizes selected children (avoid normalizing engine, Log is both engine name and log function). I'll add a comment.